### ISSUE TITLE
Nesting relationships inside arrays of objects fails with Maximum call stack size exceeded

### ISSUE THREAD

**Issue by Floriferous** [versolearning/meteor-factory#44]
The following factory creates an infinite loop:

```js
  Factory.define("book", Books, { links: [{ _id: Factory.get("author") }] });
```

The following workaround seems to work:

```js
  Factory.define("book", Books, { links: () => [{ _id: Factory.create("author")._id }] });
```

The code that loops over arrays does not use the same recursive logic as the rest of the values, but it doesn't look trivial to update.

**Comment by dburles** [comment]
Mind putting up a failing test case for that in a PR? I can take a look.

**Comment by Floriferous** [comment]
Just added one :)

**Comment by dburles** [comment]
Thanks @Floriferous. This is now fixed in 1.5.0, (see: #46).
